Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -667,6 +680,9 @@ void paint_text_fragment(PaintContext& context, Layout::TextNode const& text_nod
 
         auto text = text_node.text_for_rendering();
 
+        if (should_paint_to_alpha_mask)
+            painter.push_alpha_mask_id(foreground_text_alpha_mask_id.value());
```

Comment:
Yeah we could get rid of the push/pop calls in this part of the code by saving it in the `DrawGlyphRun` command, but we still need them for the non-glyph commands that contribute to painting the background layer. Otherwise I'd need to store the alpha mask id in those commands too, which means modifying a bunch of functions, all to get a different painter, I went with the push/pop state to avoid that and support all commands straight away.